contracts/products/SoteriaCoverageProduct.sol

```diff
@@ -130,55 +137,57 @@ contract SoteriaCoverageProduct is ISoteriaCoverageProduct, ERC721, EIP712, Reen
+        require(msg.value + _accountBalanceOf[policyholder_] > ( _maxRateNum * _chargeCycle * coverLimit_ ) / _maxRateDenom, "insufficient deposit for minimum required account balance");
 
         // deposit funds
         _deposit(policyholder_, msg.value);
 
+        // mint policy if doesn't currently exist
         if (policyID == 0) {
             policyID = ++_totalPolicyCount;
-            _ownerToPolicy[policyholder_] = policyID;
-            _policyToOwner[policyID] = policyholder_;
+            _policyOf[policyholder_] = policyID;
             _mint(policyholder_, policyID);
         }
 
         // update cover amount
-        _activeCoverAmount += coverAmount_;
-        _coverAmountOf[policyID] = coverAmount_;
+        _activeCoverLimit += coverLimit_;
+        _coverLimitOf[policyID] = coverLimit_;
        
         // update policy manager active cover amount
-        _updatePolicyManager(_activeCoverAmount);
+        _updatePolicyManager(_activeCoverLimit); // Need to change to _updateRiskManager(_activeCoverLimit)
         emit PolicyCreated(policyID);
         return policyID;
     }
 
     /**
-     * @notice Updates the cover amount of the policy.
-     * @param newCoverAmount_ The new value to cover in **ETH**.
+     * @notice Updates the cover amount of the policy, either governance or policyholder can do this.
+     * @param policyID_ The policy ID to update.
+     * @param newCoverLimit_ The new value to cover in **ETH**.
     */
-    function updateCoverAmount(uint256 newCoverAmount_) external override nonReentrant whileUnpaused {
-        require(newCoverAmount_ > 0, "zero cover value");
-        uint256 policyID = policyByOwner(msg.sender);
-        require(policyID > 0, "invalid policy");
-        uint256 currentCoverAmount = coverAmountOf(policyID);
-        require(assessRisk(currentCoverAmount, newCoverAmount_), "cannot accept that risk");
-        uint256 coverAmount = activeCoverAmount();
-        coverAmount = coverAmount + newCoverAmount_ - currentCoverAmount;
-        _coverAmountOf[policyID] = newCoverAmount_;
-        _activeCoverAmount = coverAmount;
-        _updatePolicyManager(coverAmount);
-        emit PolicyUpdated(policyID);
+    function updateCoverLimit(uint256 policyID_, uint256 newCoverLimit_) external override nonReentrant whileUnpaused {
+        require(newCoverLimit_ > 0, "zero cover value");
+        require(_exists(policyID_), "invalid policy");
+        address policyOwner = ownerOf(policyID_);
+        require(this.governance() == msg.sender || policyOwner == msg.sender, "Not owner or governance");
+        uint256 currentCoverLimit = coverLimitOf(policyID_);
+        require(_canPurchaseNewCover(currentCoverLimit, newCoverLimit_), "insufficient capacity for new cover");
+        require(_accountBalanceOf[policyOwner] > ( _maxRateNum * _chargeCycle * newCoverLimit_ ) / _maxRateDenom, "insufficient deposit for minimum required account balance");
```

Comment:
mentioned above about extracting the statement to the function

Reply:
done